### DISCUSSION TITLE
Service Token with delete database permission

### DISCUSSION THREAD

**Discussion by lucj** [planetscale/discussion#565] [Q&A] [answered]
Hello all, is there a way for a service token to get the "delete database" permission. For a project I need to automate the creation / deletion of databases, so I need to be able to do both account with the same service token.
Thanks a lot for you help.

**Comment by lucj** [comment]
Quick update. From what I saw the "delete database" permission can be given to the service token once the database is created. So I guess I can use the cli to update the current service token permission but still not able to get it right :(

When I list the service tokens it does not find my organization (which is strange as the service token I use to create the db in that same org is working fine):
```
% pscale service-token list
Error: organization MY_ORG_NAME does not exist
```
I'm not sure what I'm doing wrong here :( Any hints ?

**Comment by mscoutermarsh** [comment]
Hi @lucj,

I just looked into this and the core problem is that we don't have a `delete database` (org level) permission for Service Tokens. I can see how that'd be useful though, we'll look into if we can add it.

For:
```
Error: organization MY_ORG_NAME does not exist
```

The problem there is that service tokens are unable to access the service token endpoint themselves. That's a poor error message on our part (the request is failing, so the CLI is assuming it's due to the org missing).

**Comment by mscoutermarsh** [comment]
I've opened https://github.com/planetscale/cli/pull/758 to improve the `pscale` error message.

**Comment by mscoutermarsh** [comment]
@lucj Good news! We actually do have a `delete_databases` permission available. It's just missing from the form in the UI.

We won't have that fixed until Monday. If you're working on this over the weekend and would like to get unblocked. Email me: `mike` at planetscale dot com. And I'll see if I can get you access earlier.

**Reply by mscoutermarsh** [reply, marked as answer]
@lucj Update on this, `delete_databases` is now available in the UI.
